### PR TITLE
more rollback of chat logic

### PR DESCRIPTION
**Pull request by chrisnojima**
see if this changes anything
@mmaxim 

- [x] revert more chat logic to the older way
- [x] fix navving from profile to chat before you load the chat tab
- [x] fix closing profile card popup on mobile when going to a chat
- [x] fix ipad chat inbox search style
- [x] undo hermes

### CODE REVIEW

**Review by mmaxim — DISMISSED** [review]

File: shared/actions/chat2/index.tsx [hunk 1]
```diff
@@ -1135,13 +1130,7 @@ function* loadMoreMessages(
   }
 
   const conversationIDKey = key
-
   const conversationID = Types.keyToConversationID(conversationIDKey)
-  if (!conversationID.length) {
```
Comment (chrisnojima) [inline shared/actions/chat2/index.tsx:1140]: this is basically a noop, its a buffer and can never be null, even if that string is non hex you get a valid buffer (and it actually has length sadly). this should never fail as we check isValidConverationID above in string form anyways

File: shared/constants/router2.tsx [hunk 2]
```diff
@@ -231,25 +235,39 @@ export const navToThread = (conversationIDKey: ConversationIDKey) => {
...
-      name: 'chatConversation',
-      params: {conversationIDKey},
-    }
-    // reuse visible route if it's the same
-    const visible = chatStack.state?.routes?.[chatStack.state?.routes?.length - 1]
-    if (visible) {
-      if (visible.name === 'chatConversation' && visible.params?.conversationIDKey === conversationIDKey) {
-        convoRoute = visible
+
+    if (ChatConstants.isSplit) {
```
Comment (chrisnojima) [inline shared/constants/router2.tsx:239]: make ipad/desktop work closer to ios

File: shared/router-v2/router.desktop.tsx [hunk 3]
```diff
@@ -8,21 +8,36 @@ import {createLeftTabNavigator} from './left-tab-navigator.desktop'
...
 import * as Common from './common.desktop'
 import {HeaderLeftCancel} from '../common-adapters/header-hoc'
 
 export const headerDefaultStyle = Common.headerDefaultStyle
 const Tab = createLeftTabNavigator()
 
+type DesktopTabs = typeof Tabs.desktopTabs[number]
+
+const tabRootsVals = Object.values(tabRoots)
+// we don't want the other roots in other stacks
```
Comment (chrisnojima) [inline shared/router-v2/router.desktop.tsx:22]: while debugging i was getting into a state where the chat inbox was on top of profile which is a thing we basically never want

**Comment by chrisnojima** [comment]
@mmaxim updated

**Review by mmaxim — APPROVED** [review]

**Pull request merged**